src/Storages/StorageMongoDB.cpp

```diff
@@ -87,6 +145,72 @@ void StorageMongoDB::connectIfNotConnected()
     }
 }
 
+ColumnsDescription StorageMongoDB::getTableStructureFromData()
+{
+    connectIfNotConnected();
+
+    MongoDBCursor cursor(database_name, collection_name, {}, {}, *connection, false);
+    FormatSettings format_settings;
```

Comment:
This shoule be used from context, we can pass it as an argument for StorageMongoDB constructor